less_users/models.py

```diff
@@ -62,19 +47,34 @@ def get_points(self):
         and this challenge makes possible to get points once for every 7 days;
         for 1/day challenge points can be gained each day"""
         points = self.challenge.points
-        if len(self.log_set.all()) > 0:
-            last_log = int(self.log_set.last().date.strftime("%Y%m%d"))
-        else:
-            last_log = 0
         frequency = self.challenge.frequency
-        today = self.todays_day_num()
-        if today - last_log > frequency:
-            return points
-        return 0
+        self.check_if_active()
+        if self.is_active:
+            if self.log_set.exists():
+                last_log = self.log_set.last().date
+                if (date_today - last_log.date()).days >= frequency:
+                    return points
+                return 0
+            else:
+                return points
+        else:
+            return 0
+
+    def days_left(self, date_today=None) -> int:
```

Comment:
nie miałaś dodać typów?

Reply:
https://towardsdatascience.com/12-beginner-concepts-about-type-hints-to-improve-your-python-code-90f1ba0ac49